src/app/base/services/record/record-apollo.service.ts

```diff
@@ -533,13 +533,26 @@ export class RecordApolloService {
   }
 
   //private
-  #tokenMapper(token) {
+  #tokenMapper(token: any, tokensLength: number) {
+    let countLineBreaks = 0;
+    if (token.value.includes("\n")) {
+      countLineBreaks = token.value.split("\n").length - 1;
```

Comment:
splitting of a string costs since an array with new strings is created.
since we are doing this per token this can be a bit much. 

fasten way I know of would be a creating a small function "countOccurrance" & looping over the string
```
function countOcc(str, search){
    let c = 0,p=-1;
    while (true){
        p = str.indexOf(search,p+1)
        if(p!= -1) c++;
        else break;
    }
    return c;
}
```

Reply:
should we add this as part of the submodules? :D maybe we would need it for something else too

Reply:
yes, sounds good :)